discordo: init at unstable-2023-04-07

Arian-D opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Arian-D committed:
maintainers: add arian-d

Arian-D committed:
discordo: init at unstable-2023-03-07

lilyinstarlight reviewed (approved):
This seems to build just fine for me on x86_64-linux (though I did not actually try to run it) and the derivation looks good

Arian-D commented:
Just a mini update on this:
- I tested it on aarch64 (non-nixos)
- Updated the commit to a more recent version.

spikespaz commented:
What about [clipboard support](https://github.com/ayn2op/discordo#linux-clipboard-support)?

Should add them to `buildInputs`.

lilyinstarlight commented:
> What about [clipboard support](https://github.com/ayn2op/discordo#linux-clipboard-support)?
> 
> Should add them to `buildInputs`.

Adding them to `buildInputs` won't do anything because they are needed at runtime. We could `wrapProgram` the main binary in the derivation to always have those in `PATH`, though

Something like:

```nix
buildGoModue rec {
  #stuff...
  nativeBuildInputs = [ makeWrapper ];
  #stuff...
  postInstall = ''
    wrapProgram $out/bin/discordo \
      --prefix PATH : ${lib.makeBinPath [ xsel wl-clipboard ]}
  '';
}
```

adamcstephens reviewed (approved):
Looks good on darwin, too. This commit is now a month behind upstream.

Result of `nixpkgs-review pr 210667` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discordo</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 210667` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discordo</li>
  </ul>
</details>

lilyinstarlight commented:
Do you think you could bump this to a newer commit? I think it's probably merge-ready

Also if you want to learn something cool for automating the updates, you can add `unstableGitUpdater` to the file args and add `passthru.updateScript = unstableGitUpdater { };` to the derivation, and then r-ryantm should occasionally pick up updates automatically and you can run the update script manually like `nix-shell maintainers/scripts/update.nix --argstr commit true --argstr package discordo`

Kranzes commented on pkgs/applications/networking/discordo/default.nix:
```diff
@@ -0,0 +1,23 @@
+  };
+
+  vendorHash = "sha256-2xTGdwfQlHgprrJ1xEBSnlDHTwNMjGnhEKN1hgVg5So=";
+
```
```suggestion

  ldflags = [ "-w" "-s" ];
  
```

Arian-D commented:
I apologize about the delay; for some reason I was not getting notification until the most recent message.
- Per spikespaz's requeset I added `xsel` and `wl-clipboard` to the definition
- I added the `ldflags` per Kranzes' suggestion
- I bumped it up to a more recent version and added the unstable updater. Thank you lilyinstarlight for the guidance and advice

adamcstephens commented:
You'll want to update the title of the PR to match the new commit description.

This also builds and runs successfully as a static binary, so I'd suggest adding `CGO_ENABLED = 0;`

lilyinstarlight commented:
> * I bumped it up to a more recent version and added the unstable updater. Thank you lilyinstarlight for the guidance and advice

So I think I didn't notice it was a Go module when I said that, you actually need to use nix-update (instead of `unstableGitUpdater`) so that the vendorHash gets updated too:

```
  passthru.updateScript = nix-update-script {
    extraArgs = [ "--version=branch" ];
  };
```

SuperSandro2000 commented on pkgs/applications/networking/discordo/default.nix:
```diff
@@ -0,0 +1,40 @@
+    hash = "sha256-FUSPQK4rB0J89s+I7yhF8SQ/Q9uygQSCF9o6ltYxOk4=";
+  };
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [ "--version=branch" ];
+  };
```
passthru is usually put right before meta

SuperSandro2000 commented on pkgs/applications/networking/discordo/default.nix:
```diff
@@ -0,0 +1,40 @@
+  '';
+
+  meta = with lib; {
+    description =
+      "A lightweight, secure, and feature-rich Discord terminal client";
```
```suggestion
    description = "Lightweight, secure, and feature-rich Discord terminal client";
```

Arian-D replied:
What formatter would you suggest? Emacs (using `nixfmt` I believe) formats long lines like that

SuperSandro2000 replied:
nixpkgs-fmt or change the line length of nixfmt to 120 or something similar to not generate such silly short lines.
